// @jasonheecs/heroku-vim/lib/commands/vim.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const command_1 = require("@heroku-cli/command");
const plugin_run_v5_1 = require("@heroku-cli/plugin-run-v5");
class AppCommand extends command_1.Command {
    async run() {
        const { flags } = this.parse(AppCommand);
        let dyno_command = flags.command || 'bash';
        let opts = {
            heroku: this.legacyHerokuClient,
            app: flags.app,
            command: `
        mkdir ~/vim
        cd ~/vim
        curl 'https://s3.amazonaws.com/bengoa/vim-static.tar.gz' | tar -xz
        export VIMRUNTIME="$HOME/vim/runtime"
        export PATH="$HOME/vim:$PATH"
        cd -
        ${dyno_command}
      `,
            showStatus: true,
            attach: true,
            notify: true
        };
        let dyno = new plugin_run_v5_1.Dyno(opts);
        await dyno.start();
    }
}
AppCommand.description = 'Install Vim and use it in a dyno';
AppCommand.flags = {
    remote: command_1.flags.remote(),
    app: command_1.flags.app({ required: true }),
    command: command_1.flags.string(({ char: 'c', description: 'command to run' }))
};
exports.default = AppCommand;
